xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/changesdoc.vm

```diff
@@ -24,10 +24,18 @@
   #set ($nextNewdocLink = "<a class='changes-arrow' href='$changesLink' title='$changesTitle'>&gt;</a>")
 #end
 #if ("$!previousOrigdocVersion" != '' && "$!previousNewdocVersion" != '')
-  #set ($previousLink = ${xwiki.getURL($tdoc, 'view', "viewer=changes&rev1=${previousOrigdocVersion}&rev2=${rev1}")})
+  #if ($origdoc.RCSVersion.isLessThan($newdoc.RCSVersion))
```

Comment:
This should probably be `isLessOrEqualThan` instead, since you don't want to start considering the "reverse order" logic when comparing 2 equal versions.

Also, since this can be tricky to follow, it might be worth to document the 2 edge cases we are addressing here:
* extension version or end/beginning of history version
* reverse order navigation, preserving the reversed direction